kie-assets-library-assets/src/main/resources/smarthouse/01/heating.dmn

```diff
@@ -0,0 +1,634 @@
+      <dmn:typeRef>number</dmn:typeRef>
+    </dmn:itemComponent>
+    <dmn:itemComponent id="_A2EF9723-1E5B-4C7B-8D57-C81039A5F44A" name="t-3h" isCollection="false">
+      <dmn:typeRef>number</dmn:typeRef>
+    </dmn:itemComponent>
+  </dmn:itemDefinition>
+  <dmn:itemDefinition id="_C4C36EF9-A5CD-4F5E-BD81-D3CB5F20F5A3" name="SmartHomeContext" isCollection="false">
+    <dmn:itemComponent id="_6C2FC6BC-9AEC-4DB7-BEA4-D434D760786C" name="sensors" isCollection="true">
+      <dmn:typeRef>SensorInput</dmn:typeRef>
+    </dmn:itemComponent>
+    <dmn:itemComponent id="_719A202F-C29A-4BC4-AB88-5CB3B23CC687" name="settings" isCollection="true">
+      <dmn:typeRef>SmartHomeSetting</dmn:typeRef>
+    </dmn:itemComponent>
+    <dmn:itemComponent id="_A4497D93-B669-4502-8753-F2B3448BDF44" name="controls" isCollection="true">
+      <dmn:typeRef>SmartHomeControl</dmn:typeRef>
+    </dmn:itemComponent>
+  </dmn:itemDefinition>
+  <dmn:itemDefinition id="_151606DE-DC77-4F44-9B3D-D9CEB2F5017E" name="SmartHomeControl" isCollection="false">
+    <dmn:itemComponent id="_C5F30661-AEC9-4E05-9D43-9D2E255EE943" name="state" isCollection="false">
+      <dmn:typeRef>string</dmn:typeRef>
+      <dmn:allowedValues kie:constraintType="enumeration" id="_9EF20992-C67C-4C28-B38B-A9BC3759D5BC">
+        <dmn:text>"ON", "OFF"</dmn:text>
+      </dmn:allowedValues>
+    </dmn:itemComponent>
+    <dmn:itemComponent id="_0D8EF650-42E3-47AD-9560-80B5B86E6789" name="type" isCollection="false">
+      <dmn:typeRef>string</dmn:typeRef>
+      <dmn:allowedValues kie:constraintType="enumeration" id="_A80AF88B-8EF3-493F-96BF-4B916430F83C">
+        <dmn:text>"HEATER", "WINDOW_BLINDS", "WINDOW_OPEN"</dmn:text>
+      </dmn:allowedValues>
+    </dmn:itemComponent>
+  </dmn:itemDefinition>
+  <dmn:itemDefinition id="_9062155B-0CD0-4028-806D-02CD47842FF6" name="SmartHomeSetting" isCollection="false">
+    <dmn:itemComponent id="_CB597DE0-48B9-483A-96FB-AFF60E2B5E6C" name="type" isCollection="false">
+      <dmn:typeRef>MeasurementCategory</dmn:typeRef>
+    </dmn:itemComponent>
+    <dmn:itemComponent id="_C0B2E226-C297-4F52-BD19-D78C72CC4B63" name="threshold_low" isCollection="false">
+      <dmn:typeRef>number</dmn:typeRef>
+    </dmn:itemComponent>
+    <dmn:itemComponent id="_DEC8D386-B121-473B-A4A7-FB29B7FAA993" name="threshold_high" isCollection="false">
+      <dmn:typeRef>number</dmn:typeRef>
+    </dmn:itemComponent>
+  </dmn:itemDefinition>
+  <dmn:itemDefinition id="_4FC67739-2C75-4605-80FD-0D8293B7BA3C" name="SensorInputList" isCollection="true">
+    <dmn:typeRef>SensorInput</dmn:typeRef>
+  </dmn:itemDefinition>
+  <dmn:itemDefinition id="_8334D1FF-CE51-412E-A11E-0D0FC0B36AD6" name="SmartHomeSettingList" isCollection="true">
+    <dmn:typeRef>SmartHomeSetting</dmn:typeRef>
+  </dmn:itemDefinition>
+  <dmn:inputData id="_106B0988-E2E4-47BD-8E3A-00DE8710280E" name="context">
```

Comment:
Maybe I would suggest to have multiple input nodes. Context sounds to me like the current state of the system. For example window is open etc. Or maybe it can contain some variables about for example mode like weekend, holiday etc. Don't know exactly.

We can have input nodes for each sensor:
- outside temperature sensor 
- inside temperature sensor
- light sensor

Maybe I'm not even sure about the "context" at all because we can have one input node for each sensor and one node with decision for each actuator.

Reply:
This is part of the attempt to keep the model behaving the same way (providing same endpoint, accepting same inputs) for various complexity of models. And allows for non-breaking extension (you just add a new "branch" in the model and pass it as part of the context input).
Though if we stabilize the kinds of inputs (sensors, user settings, other considerations) we can extract from the context into separate inputs I think.